web/ui/react-app/src/pages/rules/RulePanel.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { FC, CSSProperties, Fragment } from 'react';
+import { Rule } from '../../types/types';
+import { Link } from '@reach/router';
+import { createExpressionLink, isPresent, replaceWith } from '../../utils';
+
+interface RulePanelProps {
+  rule: Rule;
+  styles?: CSSProperties;
+  tag?: keyof JSX.IntrinsicElements;
+}
+
+export const RulePanel: FC<RulePanelProps> = ({ rule, styles = {}, tag }) => {
+  const Tag = tag || Fragment;
+  const { name, query, labels, annotations, type, duration } = rule;
+  const style = { background: '#f5f5f5', ...styles };
+  const ruleName = replaceWith(name, `"`, `'`);
+  return (
+    <Tag {...(tag ? { style } : {})}>
+      <pre className="m-0" style={tag ? undefined : style}>
+        <code>
+          <div>
+            {/* Type of the rule is either alerting or recording. */}
+            {type.replace('ing', ': ')}
+            <Link to={createExpressionLink(`ALERTS{alertname="${ruleName}"}`)}>{ruleName}</Link>
+          </div>
+          <div>
+            expr: <Link to={createExpressionLink(query)}>{query}</Link>
+          </div>
+          {isPresent(duration) && <div>for: {duration}</div>}
```

Comment:
We also want to omit the `for:` field when it's present, but the value is 0.

(see https://github.com/prometheus/prometheus/blob/17d8c49919d4f26c422745b7a0a0ae5be4d63b72/pkg/rulefmt/rulefmt.go#L116, it has the `omitempty` tag set, so normal YAML rendering would omit it when it has its Go zero value (0))

Reply:
Ok, will check for zero value